s3/s3_operations.cpp

```diff
@@ -245,20 +245,51 @@ namespace irods_s3 {
             }
         }
 
-
-        // if this is a replication and we're the destination, get the data size from the source dataObjInfo
+        // special treatment for replication
+        // 1) data_size is only available from the REPLICATE_SRC entry so use that.
+        // 2) number_of_threads is available in REPLICATE_DEST entry so use that.
         if (oprType == REPLICATE_DEST) {
+            bool found_data_size = false;
+            bool found_number_of_threads = false;
 
             for (int i = 0; i < NUM_L1_DESC; ++i) {
+                const auto& l1d = L1desc[i];
+                const auto* dobj_input = l1d.dataObjInp;
+                const auto* dobj_info = l1d.dataObjInfo;
+
+                if (l1d.inuseFlag && dobj_input && dobj_input->objPath == file_obj->logical_path()) {
```

Comment:
If this was inverted, you could use `continue;` and save an indent.

Feel free to ignore this!

Reply:
Done